src/generic.rs

```diff
@@ -159,9 +159,73 @@ where
     }
 
     #[inline]
-    fn capacity(&self) -> Option<U::NonZero> {
+    fn get_capacity(&self) -> Option<U::NonZero> {
         U::maybe(self.fat, self.cap)
     }
+
+    /// Returns a pointer to underlying data
+    #[inline]
+    pub const fn as_ptr(&self) -> NonNull<T::PointerT> {
+        self.ptr
+    }
+}
+
+impl<'a, T> Cow<'a, T, Wide>
+where
+    T: Beef + ?Sized,
+{
+    /// Returns `true` if the contained data is empty
+    pub const fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    /// Returns length of underlying data
+    #[inline]
+    pub const fn len(&self) -> usize {
+        self.fat
+    }
+
+    /// Returns capacity of underlying data
+    #[inline]
+    pub const fn capacity(&self) -> Option<usize> {
+        match self.cap {
+            Some(_) => unsafe {
+                // SAFETY: Transmute from Option<NonZero<usize>> to usize is
+                // sound by definition.
+                Some(core::mem::transmute::<
+                    Option<core::num::NonZero<usize>>,
+                    usize,
+                >(self.cap))
+            },
+            None => None,
+        }
+    }
+}
+impl<'a, T> Cow<'a, T, Lean>
+where
+    T: Beef + ?Sized,
+{
+    /// Returns `true` if the contained data is empty
+    pub const fn is_empty(&self) -> bool {
+        self.len() == 0
+    }
+
+    /// Returns length of underlying data
+    #[inline]
+    pub const fn len(&self) -> usize {
+        Lean::mask_len(self.fat)
+    }
+
+    /// Returns capacity of underlying data
+    #[inline]
+    pub const fn capacity(&self) -> Option<usize> {
+        let cap = Lean::mask_cap(self.fat);
+        if cap == 0 {
+            None
+        } else {
+            Some(cap)
+        }
+    }
 }
```

Comment:
After using my version with `as_ptr` I realized it's only useful if one has externally stored length (in my case I took a string until separator). But if not, just the pointer without length doesn't help with constructing strings in const context.

I added these additional functions to allow access to all information. I'm not 100% familiar with the library internals so let me know whether they are correct.

Reply:
Specifically, whether `Lean` capacity is correct - is it 0 for borrowed values?

Added `mask_cap` does `(value & MASK_HI) >> 4`.